test/src/tests-config-api-client.ts

```diff
@@ -0,0 +1,182 @@
+                    JSON.stringify({ appName: 'Test App', kitConfigs: [] }),
+                ]);
+
+                window.mParticle.init(apiKey, config);
+
+                const configAPIClient = new ConfigAPIClient();
+                configAPIClient.getSDKConfiguration(
+                    apiKey,
+                    config,
+                    sdkInitCompleteCallback,
+                    window.mParticle.getInstance()
+                );
+
+                expect(sdkInitCompleteCallback.getCalls().length).to.equal(1);
+                expect(
+                    sdkInitCompleteCallback.calledWithMatch(
+                        apiKey,
+                        MPConfig,
+                        window.mParticle.getInstance()
+                    )
+                );
+                expect(mockServer.lastRequest).to.haveOwnProperty('response');
+
+                // TODO: Check mockServer
+                expect(mockServer.lastRequest.url).to.equal(urls.config);
+
+                const configResponse: IConfigResponse = JSON.parse(
+                    mockServer.lastRequest.response
+                );
+
+                expect(configResponse).to.be.ok;
+                expect(configResponse.appName).to.equal('Test App');
+                expect(configResponse.kitConfigs).to.be.ok;
+            });
+
+            it('should continue sdk init if fetch fails', () => {
+                const config = { requestConfig: true };
+
+                mockServer.respondWith(urls.config, [400, {}, '']);
+
+                window.mParticle.init(apiKey, window.mParticle.config);
+
+                const configAPIClient = new ConfigAPIClient();
+                configAPIClient.getSDKConfiguration(
+                    apiKey,
+                    config,
+                    sdkInitCompleteCallback,
+                    window.mParticle.getInstance()
+                );
+
+                expect(sdkInitCompleteCallback.getCalls().length).to.equal(1);
+                expect(
+                    sdkInitCompleteCallback.calledWithMatch(
+                        apiKey,
+                        MPConfig,
+                        window.mParticle.getInstance()
+                    )
+                );
+                expect(mockServer.lastRequest).to.haveOwnProperty('response');
+                expect(mockServer.lastRequest.url).to.equal(urls.config);
+            });
+        });
+
+        describe('with Data Plan', () => {
+            beforeEach(() => {
+                configUrl = `${urls.config}&plan_id=test_data_plan&plan_version=42`;
+
+                dataPlan = {
+                    planId: 'test_data_plan',
+                    planVersion: 42,
+                };
+
+                mockServer.respondWith(configUrl, [
+                    200,
+                    {},
+                    JSON.stringify({
+                        appName: 'Test App',
+                        kitConfigs: [],
+                        dataPlan,
```

Comment:
This is actually returned as a `dataPlanResult`, which I linked you [in slack to](https://mparticle.slack.com/archives/DHWS0FJTA/p1674060642699989)

Reply:
I think I'm missing something here, since the relevant code on [lines 118-127](https://github.com/mParticle/mparticle-web-sdk/blob/refactor/SQDSDKS-5010-config-api-client/src/configAPIClient.ts#L118) reference `dataPlan`, not `dataPlanResult`